apps/extension/src/ui/domains/Staking/hooks/bittensor/useCanStakeBittensor.ts

```diff
@@ -0,0 +1,59 @@
+import { ChainId } from "extension-core"
+import { useEffect, useMemo, useState } from "react"
+
+import { ScaleApi } from "@ui/util/scaleApi"
+
+import { useGetLatestBlockNumber } from "../../shared/useGetLatestBlockNumber"
+import { useGetBittensorTotalHotkeyColdkeyStakes } from "./useGetBittensorTotalHotkeyColdkeyStakes"
+import { useGetBittensorUnbondBlockNumber } from "./useGetBittensorUnbondBlockNumber"
+
+const STAKE_INTERVAL_BLOCKS = 360
+
+type CanStakeBittensor = {
+  sapi: ScaleApi | undefined | null
+  address: string | null | undefined
+  hotkey: string | number | undefined | null
+  chainId: ChainId | undefined
+}
+
+export const useCanStakeBittensor = ({ sapi, address, hotkey, chainId }: CanStakeBittensor) => {
+  const [canStake, setCanStake] = useState<boolean>(true)
+
+  const { data: blockNumber, isLoading: isBlockNumberLoading } = useGetLatestBlockNumber({
+    sapi,
+    isEnabled: chainId === "bittensor",
+  })
+
+  const { data: unbondBlockNumber } = useGetBittensorUnbondBlockNumber({
+    address,
+    delegator: hotkey,
+  })
+
+  const { data: stakeData, isLoading: isStakeDataLoading } =
+    useGetBittensorTotalHotkeyColdkeyStakes({
+      isEnabled: chainId === "bittensor",
+      sapi,
+      address: address,
+      hotkey: hotkey,
+    })
+
+  const [, lastStakedBlockNumber] = useMemo(() => stakeData ?? [0n, 0n], [stakeData])
+
+  useEffect(() => {
+    if (chainId !== "bittensor") return
+    const latestActionBlock = Math.max(
+      Number(unbondBlockNumber ?? 0),
+      Number(lastStakedBlockNumber ?? 0),
+    )
+    const coolDownBlockNumber = Number(latestActionBlock) + STAKE_INTERVAL_BLOCKS
```

Comment:
```suggestion
    const coolDownBlockNumber = latestActionBlock
      ? Number(latestActionBlock) + STAKE_INTERVAL_BLOCKS
      : 0
```

Reply:
This prevents a flash of the `canStake === false` state when the screen first loads